### PR TITLE
Revert Python finding logic and some CMake fixes for Windows

### PR DESCRIPTION
**Pull request by cyyever**
### Description

Revert the python finding logic introduced by #6381 due to unable to handle Conda builds. However, the Python executable detection was improved. In addition, some Windows warnings were re-enabled, and the hidden visibility of onnx_cpp2py_export was fixed.
### Motivation and Context
For better code.

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]

**Comment by justinchuby** [comment]
Could you explain in PR description what errors were fixed?

**Comment by cyyever** [comment]
More details were added to the PR description.

**Comment by justinchuby** [comment]
Could NOT find Python (missing: Python_INCLUDE_DIRS Python_LIBRARIES
  Interpreter Development Development.Module Development.Embed)
Call Stack (most recent call first):
  /tmp/build-env-qekth2v4/lib/python3.8/site-packages/cmake/data/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /tmp/build-env-qekth2v4/lib/python3.8/site-packages/cmake/data/share/cmake-3.30/Modules/FindPython.cmake:672 (find_package_handle_standard_args)
  CMakeLists.txt:108 (find_package)

**Comment by justinchuby** [comment]
https://github.com/onnx/onnx/actions/runs/10978690342/job/30482133230?pr=6382

**Comment by cyyever** [comment]
@justinchuby Should be fixed by passing Python_INCLUDE_DIRS from setup.py to CMake

**Comment by andife** [comment]
@cyyever I triggered more "tests" by adding the label "run release ci". 

Are the remaining failures related to the cmake fixes?

**Comment by cyyever** [comment]
Seems related. I'm investigating

**Comment by andife** [comment]
thank you

**Comment by cyyever** [comment]
@andife Help me trigger release CIs again. It's problematic that we have to remove the label and add it again to trigger them...

**Comment by cyyever** [comment]
@andife When running release CIs, the passed python_executable is incorrect, I try to read the real one from PYTHON_COMMAND env..

**Review by justinchuby — APPROVED** [review]

**Comment by justinchuby** [comment]
@andife is it possible for the workflows to be triggered by pull requests, once the label is present?

File: setup.py [hunk 1]
```diff
@@ -158,8 +158,12 @@ def run(self):
             # configure
             cmake_args = [
                 CMAKE,
-                f"-DPYTHON_INCLUDE_DIR={sysconfig.get_path('include')}",
-                f"-DPYTHON_EXECUTABLE={sys.executable}",
+                f"-DPython_INCLUDE_DIRS={sysconfig.get_path('include')}",
+                (
+                    f"-DPYTHON_EXECUTABLE={sys.executable}"
+                    if WINDOWS
+                    else f"-DPYTHON_EXECUTABLE={os.path.join(sysconfig.get_path('include'), '..', '..', 'bin', 'python3')}"
```
Comment (justinchuby) [inline setup.py:165]: Is this robust?

Comment (cyyever) [reply setup.py]: sys.executable may return a wrong path, as discussed in https://github.com/python/cpython/issues/84399

Comment (justinchuby) [reply setup.py]: I see. Could you add this as a comment?

**Comment by andife** [comment]
> @andife is it possible for the workflows to be triggered by pull requests, once the label is present?

I think I have a variant of how this could be implemented, I am currently not aware of any out-of-box variant, so I would have to delve deeper into the documentation.

**Comment by justinchuby** [comment]
> > @andife is it possible for the workflows to be triggered by pull requests, once the label is present?
> 
> I think I have a variant of how this could be implemented, I am currently not aware of any out-of-box variant, so I would have to delve deeper into the documentation.

Would something like trigger by pull request, and if the trigger is PR && label is not present, skip the workflow work?

File: setup.py (hunk 1 above)
Comment (cyyever) [reply setup.py]: Added

File: setup.py [hunk 2]
```diff
@@ -103,6 +103,24 @@ def get_ext_suffix():
     return sysconfig.get_config_var("EXT_SUFFIX")
 
 
+def get_python_execute():
+    if WINDOWS:
+        return sys.executable
+    # Try to search more accurate path, because sys.executable may return a wrong one,
```
Comment (justinchuby) [inline setup.py:109]: From the issue, sys.executable is only wrong when python is executed from gdb. Does it matter here since it is in setup.py?

Comment (cyyever) [reply setup.py:109]: It's obvious wrong when running release workflows, that is why this error was found.

**Review by justinchuby — APPROVED** [review]
Thanks!

**Pull request merged**